### PR TITLE
Switch database to MariaDB 10.6

### PR DESCRIPTION
**Pull request by Brutus5000**

### CODE REVIEW

**Comment by Brutus5000** [comment]
These failures are actually surprising me :(

**Comment by Sheikah45** [comment]
Is something in the league db not compatible with mariadb?

**Comment by Brutus5000** [comment]
On prod it is working, so I would say no.

File: build.gradle [hunk 1]
```diff
@@ -247,6 +247,7 @@ dependencies {
 
   implementation("com.github.jasminb:jsonapi-converter:${jsonapiConverterVersion}")
   runtimeOnly("mysql:mysql-connector-java:${mysqlConnectorVersion}")
```
Comment (Sheikah45) [inline build.gradle]: This could possibly be removed

File: gradle.properties [hunk 2]
```diff
@@ -5,6 +5,7 @@ springBootVersion=2.5.4
 javaxInjectVersion=1
 elideVersion=5.0.11
 mysqlConnectorVersion=8.0.25
```
Comment (Sheikah45) [inline gradle.properties]: This could possibly be removed

File: src/inttest/java/com/faforever/api/league/LeagueAbstractIntegrationTest.java [hunk 3]
```diff
@@ -6,25 +6,25 @@
...
 import org.testcontainers.containers.GenericContainer;
-import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.MariaDBContainer;
 import org.testcontainers.containers.output.Slf4jLogConsumer;
 import org.testcontainers.junit.jupiter.Testcontainers;
 
 @Testcontainers
 public abstract class LeagueAbstractIntegrationTest extends AbstractIntegrationTest {
-  private static final MySQLContainer mySQLContainer;
+  private static final MariaDBContainer mariaDBContainer;
```
Comment (bukajsytlos) [inline src/inttest/java/com/faforever/api/league/LeagueAbstractIntegrationTest.java]: when we are at it, it should probably be named leagueServiceDBContainer

**Pull request merged**